helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -597,7 +611,7 @@ public ClusterConstraints getConstraint(ClusterConstraints.ConstraintType type)
    * Notify the cache that some part of the cluster data has been changed.
    */
   public void notifyDataChange(HelixConstants.ChangeType changeType) {
```

Comment:
Thanks for using AtomicBoolean, the purpose is much clearer now. But can you add more comment on why we need a lock-free condition at this case, for better reference purpose in the future

Reply:
Sure.